### PR TITLE
Retry RPC errors

### PR DESCRIPTION
**Pull request by yeetfield**
Based on an internal Slack conversation we'd prefer to retry these to be tolerant of transient network issues

### CODE REVIEW

File: rozy/src/installers/tarball.rs [hunk 1]
```diff
@@ -48,13 +48,11 @@ impl Installer for Tarball {
 
         let downloaded_file = std::fs::File::open(file.path())?;
         let reader: Box<dyn Read> = match infer::get_from_path(file.path())? {
-            Some(t) => {
```
Comment (yeetfield) [inline rozy/src/installers/tarball.rs:51]: This is just `cargo fmt`

File: rozy/src/utils.rs [hunk 2]
```diff
@@ -1,9 +1,41 @@
 use anyhow::Result;
 
+fn is_request_retryable_based_on_error(err: &reqwest::Error) -> bool {
+    if err.is_timeout() || err.is_connect() {
```
Comment (yeetfield) [inline rozy/src/utils.rs:4]: The DNS issue we saw would satisfy `err.is_connect()`

File: rozy/src/utils.rs [hunk 3]
```diff
@@ -1,9 +1,41 @@
...
+        return true;
+    }
+
+    if err.is_decode() || err.is_body() || err.is_builder() || err.is_request() || err.is_redirect()
```
Comment (yeetfield) [inline rozy/src/utils.rs:8]: There isn't great documentation for what these mean: https://docs.rs/reqwest/0.11.14/reqwest/struct.Error.html but we can adjust as needed.

Comment (yeetfield) [reply rozy/src/utils.rs:8]: `err.is_redirect()` means that the `reqwest` library gave up after following 10 (by default) redirects

**Review by cread — APPROVED** [review]
Looks great. Thanks for the fast turnaround.

**Pull request merged**